includes/base/widget-base.php

```diff
@@ -384,7 +394,8 @@ protected function get_initial_config() {
 			$config['controls'] = $this->get_stack( false )['controls'];
 			$config['tabs_controls'] = $this->get_tabs_controls();
 		}
-		if ( isset( $config['upsale_data']['upgrade_url'] ) && false === Validate_Promotion::domain_is_on_elementor_dot_com( $config['upsale_data']['upgrade_url'] ) ) {
```

Comment:
There was an acceptance criteria to only allow url's on elementor.com when a filter overrides the promotion. This conditional resets the filter return value with the default promotion url if it's invalid.

Reply:
What if the promotion is added by 3rd party addon? you can't reset their images.

Reply:
@rami-elementor Yes I see that's where you're headed. And I agree with the general direction to allow 3rd parties to use promotions on their own widgets.  Been advocating for keeping filters very standard to be kind to hosting and 3rd parties using this.

I'm just wondering how we keep 3rd parties from highjacking links on promotions for Elementor widgets.

Reply:
Good point.

Reply:
Change reverted. For now.